src/Confluent.Kafka/TopicPartitionTimestamp.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright 2016-2017 Confluent Inc., 2015-2016 Andreas Heider
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Derived from: rdkafka-dotnet, licensed under the 2-clause BSD License.
+//
+// Refer to LICENSE for more information.
+
+namespace Confluent.Kafka
+{
+    /// <summary>
+    ///     Represents a Kafka (topic, partition, timestamp) tuple.
+    /// </summary>
+    public class TopicPartitionTimestamp
+    {
+        /// <summary>
+        ///     Initializes a new TopicPartitionTimestamp instance.
+        /// </summary>
+        /// <param name="tp">
+        ///     Kafka topic name and partition.
+        /// </param>
+        /// <param name="timestamp">
+        ///     A Kafka timestamp value.
+        /// </param>
+        public TopicPartitionTimestamp(TopicPartition tp, Timestamp timestamp)
+            : this (tp.Topic, tp.Partition, timestamp)
+        {
+        }
+
+        /// <summary>
+        ///     Initializes a new TopicPartitionTimestamp instance.
+        /// </summary>
+        /// <param name="topic">
+        ///     A Kafka topic name.
+        /// </param>
+        /// <param name="partition">
+        ///     A Kafka partition.
+        /// </param>
+        /// <param name="timestamp">
+        ///     A Kafka timestamp value.
+        /// </param>
+        public TopicPartitionTimestamp(string topic, int partition, Timestamp timestamp)
```

Comment:
I agree that we want to provide easier conversion from `DateTime` to `Timestamp`.

Of the options @treziac notes, I'm definitely against implicit conversion and `TimestampType` needs to be specified so explicit conversion is out too. 

Between the static method and constructor, I'm currently favoring using a constructor because 1. it's fewer characters. 2. we only ever want to return a `Timestamp` struct - we don't need the flexibility provided by a static method. 3. It's slightly incorrectly named - `DateTime` is not the only parameter. 4. the constructor variant reads easier to me.

I'm definitely open to being convinced back to using a static method if someone wants to present a good argument, but currently think we should change the `FromDateTime` method to an additional `Timestamp` constructor.